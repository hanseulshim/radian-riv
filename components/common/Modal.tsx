import React from 'react'

type Props = {
  title: string
  children: React.ReactNode
  id?: string
  width?: number
  percent?: number
  dark?: boolean
  closeModal: () => void
}

export default function Modal({
  children,
  id,
  title,
  closeModal,
  dark,
  width,
  percent
}: Props) {
  const handleClick = e => {
    if (e.target.className === 'modal-container') {
      closeModal()
    }
  }
  return (
    <div
      className={`modal-container${dark ? ' dark' : ''}`}
      id={id}
      onClick={handleClick}
    >
      <div
        className={`form${dark ? ' dark' : ''}`}
        style={{ width: width ? width : percent ? `${percent}%` : null }}
      >
        {title && <h2 className="form-title">{title}</h2>}
        {dark ? (
          <img
            src={`${process.env.baseUrl}/images/icon_x_light.svg`}
            className={'close-form'}
            onClick={closeModal}
          />
        ) : (
          <img
            src={`${process.env.baseUrl}/images/icon_x.svg`}
            className={'close-form'}
            onClick={closeModal}
          />
        )}
        {children}
      </div>
    </div>
  )
}
