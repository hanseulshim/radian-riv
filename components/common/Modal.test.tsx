import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Modal from './Modal'

describe('Modal', () => {
  let component
  beforeAll(() => {
    const props = {
      title: 'title',
      closeModal: () => {}
    }
    component = shallow(
      <Modal {...props}>
        <div>children</div>
      </Modal>
    )
  })
  it('renders modal-container', () => {
    expect(component.find('.modal-container')).toHaveLength(1)
  })
  it('renders form', () => {
    expect(component.find('.form')).toHaveLength(1)
  })
  it('renders title', () => {
    expect(component.find('h2.form-title').text()).toEqual('title')
  })
  it('has close image', () => {
    expect(component.find('img').prop('src')).toContain('icon_x.svg')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
