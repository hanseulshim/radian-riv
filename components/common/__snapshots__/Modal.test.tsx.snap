// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
<div
  className="modal-container"
  onClick={[Function]}
>
  <div
    className="form"
    style={
      Object {
        "width": null,
      }
    }
  >
    <h2
      className="form-title"
    >
      title
    </h2>
    <img
      className="close-form"
      onClick={[Function]}
      src="undefined/images/icon_x.svg"
    />
    <div>
      children
    </div>
  </div>
</div>
`;
